### PR TITLE
postgresqlPackages.sqlite_fdw: init at 2.4.0

### PR DESCRIPTION
**Pull request by apfelkuchen6**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    mkdir -p $out/lib $out/share/postgresql/extension
+    mv *.so $out/lib
+    mv *.sql *.control $out/share/postgresql/extension
+  '';
+
+  meta = with lib; {
+    description = "SQLite Foreign Data Wrapper for PostgreSQL";
+    homepage = "https://github.com/pgspider/sqlite_fdw";
+    maintainers = with maintainers; [ apfelkuchen6 ];
+    platforms = platforms.unix;
```
Comment (FliegendeWurst) [inline pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix:37]: Please set platforms = linux or fix the Darwin build failure
```
       > Running phase: installPhase
       > mv: missing destination file operand after '/nix/store/kxzxm94b3di0acd7g1prcbfqp69s3112-sqlite_fdw-2.4.0/lib'
       > Try 'mv --help' for more information.
```

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    postgresql
+  ];
+
+  makeFlags = [ "USE_PGXS=1" ];
+
+  installPhase = ''
+    mkdir -p $out/lib $out/share/postgresql/extension
+    mv *.so $out/lib
+    mv *.sql *.control $out/share/postgresql/extension
+  '';
```
Comment (FliegendeWurst) [inline pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix:31]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/lib $out/share/postgresql/extension
    mv *.so $out/lib
    mv *.sql *.control $out/share/postgresql/extension
    runHook postInstall
  '';
```
I think you need to use *.dylib or something on Darwin.

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  meta = with lib; {
+    description = "SQLite Foreign Data Wrapper for PostgreSQL";
+    homepage = "https://github.com/pgspider/sqlite_fdw";
+    maintainers = with maintainers; [ apfelkuchen6 ];
+    platforms = platforms.unix;
+    license = licenses.postgresql;
```
Comment (FliegendeWurst) [inline pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix:38]:
```suggestion
  meta = {
    description = "SQLite Foreign Data Wrapper for PostgreSQL";
    homepage = "https://github.com/pgspider/sqlite_fdw";
    maintainers = with lib.maintainers; [ apfelkuchen6 ];
    platforms = lib.platforms.unix;
    license = lib.licenses.postgresql;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by apfelkuchen6** [comment]
@ofborg build postgresql12Packages.sqlite_fdw postgresql13Packages.sqlite_fdw postgresql14Packages.sqlite_fdw postgresql15Packages.sqlite_fdw postgresql16Packages.sqlite_fdw

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix (hunk 1 above)
Comment (apfelkuchen6) [reply pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix]: Thanks, it builds on macOS with `*.so *.dylib`. I don't have a mac to test whether it actually works though.

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]

**Review by Ma27 — CHANGES_REQUESTED** [review]
> Then remove the whole install phase, this should work out of the box

Also replace `stdenv.mkDerivation` with `buildPostgresqlExtension`.

Blocking to make sure nobody merges it before.

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  makeFlags = [ "USE_PGXS=1" ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/lib $out/share/postgresql/extension
+    mv *.so *.dylib $out/lib
+    mv *.sql *.control $out/share/postgresql/extension
+    runHook postInstall
+  '';
```
[the remark this answers is not in the document]
Comment (apfelkuchen6) [reply pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix]: done

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]

File: pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchFromGitHub,
+  sqlite,
+  postgresql,
```
[the remark this answers is not in the document]
Comment (apfelkuchen6) [reply pkgs/servers/sql/postgresql/ext/sqlite_fdw.nix:5]: The `postgresql` argument is also used to mark version 17+ as broken. Is there another way to access the postgres version used?

**Review by wolfgangwalther — APPROVED** [review]

**Review by Ma27 — APPROVED** [review]

**Pull request merged**